### ISSUE TITLE
Running flw baseline fedAVG_MNIST with recommended parameters on README gives accuracy better than the paper "best result"

### ISSUE THREAD

**Issue by cvrnogueira** [adap/flower#3235]
### Describe what the documentation is missing.

On this part https://github.com/adap/flower/tree/main/baselines/flwr_baselines/flwr_baselines/publications/fedavg_mnist there is a README suggesting to run the code with `python main.py num_epochs=5 num_rounds=1000 iid=True`. When I did using the suggested string but with  num_rounds=100 , I got a result plot with a baseline better than the "paper best result". I'm running locally in a Macbook M1 Max

### Suggest your improvement.

I think is very confusing to have received a result plot with an accuracy better than the  "paper best result". So I think an explanation on the README of `flwr_baselines/publications/fedavg_mnist ` about this case will be nice. 
<img width="500" alt="image" src="https://github.com/adap/flower/assets/20215131/470f32b9-0d39-4ec1-8761-4557184469fa">

**Comment by quang14github** [comment]
Hi @cvrnogueira! I am doing experiments with the Flower framework and can produce the same result as yours. However, when I try to run Flower with non-iid data (iid=False), the server-side evaluation is poor (the accuracy is around 0.21) although client-side evaluation can reach over 0.98. Have you faced with the same problem?
Below is my modified version of the baseline. You can also take a look at my config.
[https://github.com/quang14github/flower/tree/main/baselines/flwr_baselines/flwr_baselines/publications/fedavg_mnist](https://github.com/quang14github/flower/tree/main/baselines/flwr_baselines/flwr_baselines/publications/fedavg_mnist)

**Comment by WilliamLindskog** [comment]
Hi @cvrnogueira 

Thanks for raising this. Is this something that you are still experiencing?

**Comment by WilliamLindskog** [comment]
The discrepancy in accuracy compared to the original FedAvg paper is likely due to differences in experimental setup rather than an issue with Flower’s implementation. Here are a few key factors that could explain discrepancy:
1. Hardware Performance – Newer hardware can offer improved computational efficiency compared to the hardware used in the original FedAvg paper. Faster computations can lead to better convergence properties.
2. (Maybe) Modern Framework Optimizations – Advances in PyTorch, TensorFlow, and machine learning optimizations over time have likely contributed to improved model training dynamics compared to the original study.

It seems that your results align with expected variations due to different experimental conditions. There doesn’t appear to be an issue with Flower’s `FedAvg` implementation.

Closing this issue for now.
